pkg/os/exec.go

```diff
@@ -5,33 +5,44 @@ import (
 	"github.com/code-ready/crc/pkg/crc/logging"
 	"os"
 	"os/exec"
+	"strings"
 )
 
+func run(command string, args []string, env map[string]string) (string, string, error) {
+	cmd := exec.Command(command, args...) // #nosec G204
+	cmd.Env = ReplaceEnv(os.Environ(), "LC_ALL", "C")
+	cmd.Env = ReplaceEnv(cmd.Env, "LANG", "C")
+	if len(env) != 0 {
+		cmd.Env = os.Environ()
+		for key, value := range env {
+			cmd.Env = ReplaceEnv(cmd.Env, key, value)
+		}
+	}
+	stdOut := new(bytes.Buffer)
+	stdErr := new(bytes.Buffer)
+	cmd.Stdout = stdOut
+	cmd.Stderr = stdErr
+	logging.Debugf("Running '%s %s'", command, strings.Join(args, " "))
+	err := cmd.Run()
+	if err != nil {
+		logging.Debugf("Command failed: %v", err)
+		logging.Debugf("stdout: %s", stdOut.String())
+		logging.Debugf("stderr: %s", stdErr.String())
+	}
+	return stdOut.String(), stdErr.String(), err
+}
+
 // RunWithPrivilege executes a command using sudo
 // provide a reason why root is needed as the first argument
 func RunWithPrivilege(reason string, cmdAndArgs ...string) (string, string, error) {
 	sudo, err := exec.LookPath("sudo")
 	if err != nil {
 		return "", "", err
 	}
-	cmd := exec.Command(sudo, cmdAndArgs...) // #nosec G204
-	stdOut := new(bytes.Buffer)
-	stdErr := new(bytes.Buffer)
-	cmd.Stdout = stdOut
-	cmd.Stderr = stdErr
 	logging.Infof("Will use root access: %s", reason)
-	err = cmd.Run()
-	return stdOut.String(), stdErr.String(), err
+	return run(sudo, cmdAndArgs, map[string]string{})
 }
 
 func RunWithDefaultLocale(command string, args ...string) (string, string, error) {
-	cmd := exec.Command(command, args...) // #nosec G204
-	cmd.Env = ReplaceEnv(os.Environ(), "LC_ALL", "C")
-	cmd.Env = ReplaceEnv(cmd.Env, "LANG", "C")
-	stdOut := new(bytes.Buffer)
-	stdErr := new(bytes.Buffer)
-	cmd.Stdout = stdOut
-	cmd.Stderr = stdErr
-	err := cmd.Run()
-	return stdOut.String(), stdErr.String(), err
+	return run(command, args, map[string]string{"LC_ALL": "C", "LANG": "C"})
```

Comment:
@cfergeau in the `run` function you already using the `LC_ALL` and `LANG` env value so why we are passing this here again?

```golang
func run(command string, args []string, env map[string]string) (string, string, error) {
	cmd := exec.Command(command, args...) // #nosec G204
	cmd.Env = ReplaceEnv(os.Environ(), "LC_ALL", "C")
	cmd.Env = ReplaceEnv(cmd.Env, "LANG", "C")
...
```

Reply:
I forgot to remove these lines from `run`, thanks for spotting this!